#!/usr/bin/env python
PACKAGE = "topic_transit"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_z", double_t, 0, "Minimum height for passthrough filter", 0.5, -2.0, 2.0)
gen.add("max_z", double_t, 0, "Maximum height for passthrough filter", 0.5, -2.0, 2.0)
gen.add("max_radius", double_t, 0, "Maximum radius for outlier removal", 0.2, 0.1, 2.0)
gen.add("min_neighbors", int_t, 0, "Min neighbors for outlier removal", 10, 1, 100)
gen.add("leaf_size", double_t, 0, "Leaf size for voxel grid filter", 0.1, 0.01, 1.0)
gen.add("max_window_size", int_t, 0, "Maximum window size for PMF", 20, 5, 50)
gen.add("slope", double_t, 0, "Slope threshold for PMF", 0.2, 0.1, 2.0)
gen.add("initial_distance", double_t, 0, "Initial distance threshold for PMF", 0.1, 0.1, 1.0)
gen.add("max_distance", double_t, 0, "Maximum distance threshold for PMF", 1.0, 0.5, 2.0)

gen.add("use_ground_filter", bool_t, 0, "Enable/Disable ground filtering", True)
gen.add("publish_ground", bool_t, 0, "Enable/Disable ground cloud publishing", False)

exit(gen.generate(PACKAGE, "topic_transit", "CloudFilter"))